readthedocs/readthedocs.org
# Serve external version through El Proxito

Allow El Proxito to serve `EXTERNAL` versions.

Before this PR, we were serving external version just by using an NGINX `proxy_pass` rule pointing to the Blob storage. This new behavior is cleaner but also required to serve these versions on corporate, since they require auth and El Proxito can handle it properly.

### Steps to test this PR

Import `readthedocs/test-builds` into your local instance:

```bash
python manage.py import_project_from_live test-builds
```
Then, open a Django shell and create an external version pointing to PR 13 in that repository:

```python
project = Project.objects.get(slug='test-builds')
version = Version.objects.create(project=project, type='external', active=True, verbose_name='13', slug='13', identifier='617672ce68a24c4397ae009a69b4d93850e4b8b6')

```

> You can also setup a OAuth GitHub application, connect your GH account, etc but that's too complicated just for testing accessing to an external version documentation.

Finally, trigger a build for that version:

```python
from readthedocs.core.utils import trigger_build
trigger_build(project, version)
```

At this point, you will be able to hit http://external-builds.community.dev.readthedocs.io/html/test-builds/13/index.html and see the documentation built from a PR properly.
